### PR TITLE
Fix older bash failing on unbound array variables

### LINKED ISSUE #75929: Failed to setup cluster via local-up-cluster.sh
**What happened**:

Failed to setup cluster via `local-up-cluster.sh` in master branch.

```
use 'kubectl --kubeconfig=/var/run/kubernetes/admin-kube-aggregator.kubeconfig' to use the aggregated API server
hack/local-up-cluster.sh: line 633: node_cidr_args[@]: unbound variable
Cleaning up...
hack/local-up-cluster.sh: line 339: wait_for: No record of process 9141
...
2019/03/31 19:46:32 [WARNING] This certificate lacks a "hosts" field. This makes it unsuitable for
websites. For more information see the Baseline Requirements for the Issuance and Management
of Publicly-Trusted Certificates, v.1.1.6, from the CA/Browser Forum (https://cabforum.org);
specifically, section 10.2.3 ("Information Requirements").
/go/src/k8s.io/kubernetes/hack/lib/util.sh: line 562:  8653 Terminated              etcd --advertise-client-urls "${KUBE_INTEGRATION_ETCD_URL}" --data-dir "${ETCD_DIR}" --listen-client-urls "${KUBE_INTEGRATION_ETCD_URL}" --debug 2> "${ETCD_LOGFILE}" > /dev/null
The connection to the server localhost:6443 was refused - did you specify the right host or port?
Kube-dns addon successfully deployed.
hack/local-up-cluster.sh: line 782: cni_conf_dir_args[@]: unbound variable
Cleaning up...
```

Perhaps related with https://github.com/kubernetes/kubernetes/pull/75662.

@s-ito-ts @dims 

**What you expected to happen**:

Run `local-up-cluster.sh` without error.

**How to reproduce it (as minimally and precisely as possible)**:

./hack/local-up-cluster.sh

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mrunalp**
Fixes #75929 
Older bash versions fail when trying to expand array variables
when the array is empty.

Signed-off-by: Mrunal Patel <mpatel@redhat.com>

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -638,7 +638,7 @@ function start_controller_manager {
       --cluster-signing-cert-file="${CLUSTER_SIGNING_CERT_FILE}" \
       --cluster-signing-key-file="${CLUSTER_SIGNING_KEY_FILE}" \
       --enable-hostpath-provisioner="${ENABLE_HOSTPATH_PROVISIONER}" \
-      "${node_cidr_args[@]}" \
+      ${node_cidr_args[@]+"{node_cidr_args[@]}"} \
```
Comment (s-ito-ts) [inline hack/local-up-cluster.sh]: Thank you for fix.
I think that this code needs "$" like other changes.

Comment (mrunalp) [reply hack/local-up-cluster.sh]: Fixed. Thanks!

**Pull request merged**